tests/compile-fail/used_underscore_binding.rs

```diff
@@ -9,7 +9,15 @@ fn prefix_underscore(_foo: u32) -> u32 {
 
 /// Test that we lint even if the use is within a macro expansion
 fn in_macro(_foo: u32) {
-    println!("{}", _foo); // doesn't warn, nut should #507
+    println!("{}", _foo); //~ ERROR used binding which is prefixed with an underscore
+}
+
+// TODO: This doesn't actually correctly test this. Need to find a #[derive(...)] which sets off
```

Comment:
`#[derive(RustcEncodable)]` should work. However, you need to include `rustc_serialize`, which is a Crates.io crate, so it will not work directly in a compile-fail test. Instead, add a small test, similar to mut_mut_macro, directly in `tests/` (which uses RustcEncodable on a non-empty struct and has `#[deny(this lint)]`), and a dev-dependency on rustc-serialize in Cargo.toml. The test should always pass since this lint will not trigger there.

(Ensure the test fails when you use it for an earlier clippy version)